### PR TITLE
Refactor project design, remove duplicate code, implement temporary package structure

### PR DESCRIPTION
**Pull request by ryanw-ny**
This removes any bloated or duplicate code from the application, it also moves things around to add more organization and make it easier to work with in the future. Temporary package structure has been implemented although it will probably be revised down the road as more features get implemented.

This shouldn't impact performance or how the program currently functions, this just cleans things up. All sub-activities have been migrated to their own package and can be easily found in ```mozilla.org.webmaker.activity``` excluding the default activity and web maker activities. 

This closes issue  #1501

/cc @thisandagain

### CODE REVIEW

File: app/src/main/java/mozilla/org/webmaker/WebmakerActivity.java [hunk 1]
```diff
@@ -34,10 +34,9 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     public void onDestroy() {
         Log.v("wm", "onDestroy");
-        if (view != null) {
-            view.destroy();
-            view = null;
-        }
+        assert view != null;
```
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/WebmakerActivity.java]: This assert is removed during runtime which makes this brittle, no?

File: app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java [hunk 2]
```diff
@@ -3,12 +3,16 @@
...
+import mozilla.org.webmaker.activity.Editor;
+import mozilla.org.webmaker.activity.Map;
+import mozilla.org.webmaker.activity.Project;
+import mozilla.org.webmaker.activity.Tinker;
 import mozilla.org.webmaker.router.Router;
 
 public class WebmakerApplication extends Application {
 
     // Singleton
-    private static WebmakerApplication singleton;
```
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java]: Good catch. :+1:

File: app/src/main/java/mozilla/org/webmaker/fragment/WebviewFragment.java [hunk 3]
```diff
@@ -68,34 +44,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
     @Override
     public void onDestroyView() {
-        if (mWebView != null) {
-            mWebView.destroy();
-            mWebView = null;
-        }
+        assert mWebView != null;
```
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/fragment/WebviewFragment.java]: Same question here.

**Comment by thisandagain** [comment]
Looks great. Just one small question / concern regarding the use of an assertion within the `onDestroyView` lifecycle event.

**Comment by ryanw-ny** [comment]
@thisandagain Good catch with the assertions! I went ahead and replaced all of them with regular conditional statements like suggested and added a check to see if the ActionBar is null and if so throw a error for debugging purposes.

File: app/src/main/java/mozilla/org/webmaker/MainActivity.java [hunk 4]
```diff
@@ -29,7 +29,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // Set up the action bar.
         final ActionBar actionBar = getActionBar();
-        assert actionBar != null;
+        if (actionBar == null) throw new NullPointerException("ActionBar has returned null!");
```
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/MainActivity.java]: :+1:

**Comment by thisandagain** [comment]
Awesome. Looks great. R+ :+1: :cat2: :cake: :boom:

**Pull request merged**